vxfreeswitch/voice.py

```diff
@@ -131,6 +132,45 @@ def unboundEvent(self, evdata, evname):
         log.msg("Unbound event %r" % (evname,))
 
 
+class FreeSwitchESLClientProtocol(FreeSwitchESLProtocol):
+    def __init__(self, vumi_transport):
+        EventProtocol.__init__(self)
+        self.vumi_transport = vumi_transport
+        self.job_queue = {}
+        self.ready = Deferred()
+
+    @inlineCallbacks
+    def connectionMade(self):
+        yield self.eventplain("BACKGROUND_JOB CHANNEL_HANGUP")
+        self.ready.callback(self)
+
+    def make_call(self, number):
+        def _success(ev):
+            response = Deferred()
+            self.job_queue[ev.Job_UUID] = response
+            return response
+        self.uniquecallid = number
+        profile = self.vumi_transport.config.sofia_profile
+        call_url = "sofia/%s/%s" % (profile, number)
+        d = self.bgapi("originate %s" % (call_url))
+        d.addCallback(_success)
+        return d
+
+    def onBackgroundJob(self, ev):
+        d = self.job_queue.pop(ev.Job_UUID, None)
+        if d:
+            response, content = ev.rawresponse.split()
+            if response == "+OK":
+                d.callback(content)
```

Comment:
This is handled in https://github.com/praekelt/vumi-freeswitch-esl/pull/12 (it's a bit messy now, but will clean up when this PR lands)